common/request_test.go

```diff
@@ -51,7 +52,26 @@ func TestRequest(t *testing.T) {
 		Timestamp: &ts,
 		WallTime:  &wt,
 	}
-	req := NewRequest(ctx, evt, nil, nil, "intercept", false)
+	req, err := NewRequest(ctx, evt, nil, nil, "intercept", false)
+	require.NoError(t, err)
+
+	t.Run("error_parse_url", func(t *testing.T) {
+		t.Parallel()
+		evt := &network.EventRequestWillBeSent{
+			RequestID: network.RequestID("1234"),
+			Request: &network.Request{
+				URL:      ":",
+				Method:   "POST",
+				Headers:  network.Headers(headers),
+				PostData: "hello",
+			},
+			Timestamp: &ts,
+			WallTime:  &wt,
+		}
+		req, err := NewRequest(ctx, evt, nil, nil, "intercept", false)
+		require.EqualError(t, err, `cannot parse URL: parse ":": missing protocol scheme`)
```

Comment:
It's fine this way, and I think we don't need to look for an exact match here. It's fine to test if the error just contains `cannot parse URL`.